engine/language-server/src/main/scala/org/enso/languageserver/boot/TimingsConfig.scala

```diff
@@ -39,5 +39,5 @@ class TimingsConfig(
 }
 
 object TimingsConfig {
-  def default(): TimingsConfig = new TimingsConfig(10.seconds)
+  def default(): TimingsConfig = new TimingsConfig(20.seconds)
```

Comment:
I don't think increasing time fixes anything, it only hides the problem behind a carpet. We may want to add this just before the release (only on a release branch) as a temporary mitigation of problems we don't have a solution for, but such a change doesn't belong into future versions of Enso.

We should investigate what is happening in the engine when the timeout exceeds 10s.

NetBeans used [fitness via postmortem](https://netbeans.apache.org/wiki/FitnessViaPostMortem.html) system and it would perfectly fit here as well. When the reply takes longer (for example more than 10s), start automatic profiling of the remaining activity. When the hard timeout (for example 20s) is really hit, dump the profiling data.

NetBeans used 1s/3s for development builds and 10s/30s for production builds as soft/hard time limits.

Reply:
What about computational operations that really do take that long? Processing a huge amount of data, doing some long-running computation?